coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,690 @@
+                    groups[variable] = (None
+                                        if groups.get(variable, "") == "" else
+                                        int(groups[variable]))
+
+                if "origin" in groups:
+                    groups["origin"] = "{} ({})".format(
+                        str(klass.__name__),
+                        str(groups["origin"]))
+
+                # Construct the result.
+                return Result.from_values(
+                    origin=groups.get("origin", self),
+                    message=groups.get("message", ""),
+                    file=filename,
+                    severity=int(groups.get("severity",
+                                            RESULT_SEVERITY.NORMAL)),
+                    line=groups["line"],
+                    column=groups["column"],
+                    end_line=groups["end_line"],
+                    end_column=groups["end_column"])
+
+            def process_output_corrected(self,
+                                         output,
+                                         filename,
+                                         file,
+                                         diff_severity=RESULT_SEVERITY.NORMAL,
+                                         diff_message="Inconsistency found."):
+                """
+                Processes the executable's output as a corrected file.
+
+                :param output:
+                    The output of the program. This can be either a single
+                    string or a sequence of strings.
+                :param filename:
+                    The filename of the file currently being corrected.
+                :param file:
+                    The contents of the file currently being corrected.
+                :param diff_severity:
+                    The severity to use for generating results.
+                :param diff_message:
+                    The message to use for generating results.
+                :return:
+                    An iterator returning results containing patches for the
+                    file to correct.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for diff in Diff.from_string_arrays(
+                                file,
+                                string.splitlines(keepends=True)).split_diff():
+                        yield Result(self,
+                                     diff_message,
+                                     affected_code=(diff.range(filename),),
+                                     diffs={filename: diff},
+                                     severity=diff_severity)
+
+            def process_output_regex(
+                    self,
+                    output,
+                    filename,
+                    file,
+                    output_regex,
+                    severity_map=MappingProxyType({
```

Comment:
You could actually do a tuple of tuples:

```
In [30]: dict((("a", "b"), ("c", "d")))
Out[30]: {'a': 'b', 'c': 'd'}
```
And do `dict()` inside the function. This would convert any derived type into a `dict` and throw a TypeError when it can't - which makes it easy to read + flexible IMO